graphql-dgs/src/test/kotlin/com/netflix/graphql/dgs/internal/SourceArgumentTest.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2025 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs.internal
+
+import com.netflix.graphql.dgs.DgsComponent
+import com.netflix.graphql.dgs.DgsData
+import com.netflix.graphql.dgs.DgsQuery
+import com.netflix.graphql.dgs.Source
+import com.netflix.graphql.dgs.internal.method.DataFetchingEnvironmentArgumentResolver
+import com.netflix.graphql.dgs.internal.method.FallbackEnvironmentArgumentResolver
+import com.netflix.graphql.dgs.internal.method.InputArgumentResolver
+import com.netflix.graphql.dgs.internal.method.MethodDataFetcherFactory
+import com.netflix.graphql.dgs.internal.method.SourceArgumentResolver
+import graphql.GraphQL
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.boot.test.context.runner.ApplicationContextRunner
+import org.springframework.context.ApplicationContext
+import java.util.*
+
+internal class SourceArgumentTest {
+    private val contextRunner = ApplicationContextRunner()
+
+    private fun schemaProvider(applicationContext: ApplicationContext) =
+        DgsSchemaProvider(
+            applicationContext = applicationContext,
+            federationResolver = Optional.empty(),
+            existingTypeDefinitionRegistry = Optional.empty(),
+            schemaLocations = listOf("source-argument-test/schema.graphqls"),
+            methodDataFetcherFactory =
+                MethodDataFetcherFactory(
+                    listOf(
+                        InputArgumentResolver(DefaultInputObjectMapper()),
+                        DataFetchingEnvironmentArgumentResolver(applicationContext),
+                        SourceArgumentResolver(),
+                        FallbackEnvironmentArgumentResolver(DefaultInputObjectMapper()),
+                    ),
+                ),
+        )
+
+    data class Show(
+        val title: String,
+    )
+
+    @Test
```

Comment:
Depending on whether @Source can be null, consider adding a test case for that scenario.